Formula/dbus.rb

```diff
@@ -32,38 +32,50 @@ class Dbus < Formula
 
   uses_from_macos "expat"
 
-  on_linux do
-    depends_on "pkg-config" => :build
+  on_macos do
+    # Patch applies the config templating fixed in https://bugs.freedesktop.org/show_bug.cgi?id=94494
+    # Homebrew pr/issue: 50219
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/0a8a55872e/d-bus/org.freedesktop.dbus-session.plist.osx.diff"
+      sha256 "a8aa6fe3f2d8f873ad3f683013491f5362d551bf5d4c3b469f1efbc5459a20dc"
+    end
   end
 
-  # Patch applies the config templating fixed in https://bugs.freedesktop.org/show_bug.cgi?id=94494
-  # Homebrew pr/issue: 50219
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/0a8a55872e/d-bus/org.freedesktop.dbus-session.plist.osx.diff"
-    sha256 "a8aa6fe3f2d8f873ad3f683013491f5362d551bf5d4c3b469f1efbc5459a20dc"
+  on_linux do
+    depends_on "pkg-config" => :build
   end
 
   def install
     # Fix the TMPDIR to one D-Bus doesn't reject due to odd symbols
     ENV["TMPDIR"] = "/tmp"
 
-    # macOS doesn't include a pkg-config file for expat
-    ENV["EXPAT_CFLAGS"] = "-I#{MacOS.sdk_path}/usr/include"
-    ENV["EXPAT_LIBS"] = "-lexpat"
+    on_macos do
+      # macOS doesn't include a pkg-config file for expat
+      ENV["EXPAT_CFLAGS"] = "-I#{MacOS.sdk_path}/usr/include"
+      ENV["EXPAT_LIBS"] = "-lexpat"
+    end
```

Comment:
We shouldn't need this if you make `pkg-config` a build dependency on macOS too. I think taking out two `on_os` blocks makes the added dependency worth it.

Reply:
I tested this locally and adding `pkg-config` as a build dependency for both platforms did allow me to remove those lines.  This really cleans up the formula nicely!